scripts/apps/workspace/content/directives/ContentProfileSchemaEditor.ts

```diff
@@ -238,7 +247,17 @@ export function ContentProfileSchemaEditor(content, metadata, vocabularies) {
                         scope.model.schema[schema.key] = {enabled: true};
                     }
 
-                    schemaKeys.splice(position === 'before' ? order - 1 : order + 1, 0, schema.key);
+                    if (scope.sectionToRender === 'content') {
```

Comment:
Why is it important which section it is? Why does the header section work correctly without modification?

Reply:
We have different `schemaKeys` for `header fields` and `content fields` and for `content fields` the `order = no. of header fields + index in schemaKeys for content fields`. So, if header fields are `[h1, h2]` and content fields are `[c1, c2]` and I want to add a new content field before `c2` then  `order = 4, position = 'before'` and `schemaKeys.length = 2` for content fields. So it will push the `c3` at the end of schemaKeys of content fields like`[c1, c2, c3]` but expected result should be like this` [c1, c3, c2]`. And such is not the case with header fields as both `order` as well as length of schemaKeys starts from 1.

Reply:
The way we are handling the order looks messy and harder to understand on every modification. Do you think you could refactor it so the same code could work on both groups with no exceptional handling?

Reply:
I will try to do that.